main/lsp/updates.cc

```diff
@@ -265,6 +265,9 @@ bool LSPLoop::canTakeFastPath(const FileUpdates &updates, const vector<core::Fil
             } else {
                 auto &oldHash = globalStateHashes[fref.id()];
                 ENFORCE(oldHash.definitions.hierarchyHash != core::GlobalStateHash::HASH_STATE_NOT_COMPUTED);
+                if (hashes[i].definitions.hierarchyHash == core::GlobalStateHash::HASH_STATE_INVALID) {
+                    return false;
+                }
                 if (hashes[i].definitions.hierarchyHash != core::GlobalStateHash::HASH_STATE_INVALID &&
```

Comment:
I think you can just change this line to `==`.

Reply:
Or remove the first conditional here and just have two separate checks and two separate logger messages, e.g.

```c++
if (HASH_STATE_INVALID) {
  logger->debug("Taking slow path because {} has a syntax error");
} else if (new != old) {
  // ...
}
```